### DISCUSSION TITLE
BBS02B original firmware

### DISCUSSION THREAD

**Discussion by enduring78** [danielnilsson9/bbs-fw#94] [Q&A]
Hi everyone, 

I'm considering flashing this firmware because of the street legal option. The instructions say to have a copy of the original firmware handy. It might be a silly question, but how do I read the firmware, or where can I get it? 

I have looked online but I can only find other modded firmwares, no original factory firmware. 

Help is much appreciated!

EDIT: I have gone ahead and flashed the firmware anyway because of the 'standard' and 'sport' modes. I would still like to have the original firmware though!

**Comment by NeoCHoZen** [comment]
Did you do a backup (save as when launching the application used to flash)? I didn't do that and lost the original firmware. I would have liked to go back to but can't now. So the standard and sport modes are just different assist levels (assuming you have 9). I believe PAS 1, 3, 5, 7 and 9 are the "Standard" and 2,4,6,8 and 9 are the "Sport" with 9 being level 5 in both modes. It might be confusing at first. The only hard part is not knowing what the original settings were unless you did screen shots (I didn't do that either). I guess it's the price of being impatient. I get it.

**Reply by enduring78** [reply]
Hi, no, I did not make a backup of the stock firmware, nor have I found a way to do it. I did back up the settings set the the Bafang Configurator though.

I have a display that has all 9 modes, and I do not recognize what you are saying at all. It's like I can switch to a totally different bike when off-roading. Maybe you're thinking you activated sport mode but actually just stayed on standard?

**Comment by e-i-k-e** [comment]
@NeoCHoZen 
How can the backup be created when the application for bottling is started?

**Comment by dav0000000** [comment]
It is not possible to create a backup of the firmware on your controller. You should be able to find the original firmware with a quick search but this firmware is better.

**Reply by enduring78** [reply]
I have searches for the original firmware many times, an did could not find any. Can you please send it to me if you can find it?

Off course this firmware is much better, but if I ever need to sell this motor or send it off for a RMA, I'd rather not have them see this firmwae.

**Comment by e-i-k-e** [comment]
Hello, I couldn't find anything either.  
I would be happy if someone would provide the firmware (incl. BBS02).

**Reply by Gonespa** [reply]
Take a look at this thread: https://endless-sphere.com/sphere/threads/bbs02-firmware-required.113107/
Bear in mind I did not test that and apparently someone had problems with it.

**Reply by Gonespa** [reply]
Specifically this: https://support.biktrix.com/kb/guide/en/52v-programming-for-bbs02-motor-K2gUq2nWAX/Steps/1132765  but apparently is for 52v batteries only

**Reply by Gonespa** [reply]
And check the thread named BBS02. You will find this same information

**Reply by e-i-k-e** [reply]
Thanks!
